cernopendata_client/searcher.py

```diff
@@ -196,51 +196,34 @@ def get_files_list(
     if server != SERVER_HTTP_URI and searcher_protocol != "xrootd":
         searcher_protocol = server.split(":")[0]
     files_list = []
+
+    new_server = SERVER_ROOT_URI
+    if searcher_protocol == "http":
+        new_server = server
+    elif searcher_protocol == "https":
+        new_server = SERVER_HTTPS_URI
+
     for file_ in record_json["metadata"]["files"]:
         files_list.append((file_["uri"], file_["size"], file_["checksum"]))
-    if expand:
-        # let's unwind file indexes
-        files_list_expanded = []
-        for file_ in files_list:
-            if file_[0].endswith("_file_index.json"):
-                try:
-                    url_file = "{}/record/{}/files/{}".format(
-                        server, str(record_json["id"]), file_[0].split("/")[-1]
+    for file_ in record_json["metadata"]["_file_indices"]:
+        if expand:
+            # let's unwind file indexes
```

Comment:
Note that the changes do not pass unit tests, e.g. see the CI report for Python 3.12:

```
================== 22 failed, 50 passed, 8 skipped in 45.12s ===================
```